### DISCUSSION TITLE
Restore from backup procedure?

### DISCUSSION THREAD

**Discussion by krestian83** [Security-Onion-Solutions/securityonion#7859] [General]
Hello,

I can not find a documented procedure for restoring main/master node from backup?  I have seen the documentation here: https://docs.securityonion.net/en/2.3/backups.html but to me it is not self explanatory what to do.

untar to original locations like this: `tar -xvf backup.tar -C /`   and restart node?

**Comment by tomjones1977** [comment]
YMMV, but viewing the contents of the tar file look like it was backed up at the root directory, so in theory your command should work.  However I've never tested this in anger so I don't know for definite that it will work as expected.

If one of the maintainers can chime in I'm happy to write some restore docs and submit a MR?

**Comment by cm-ops** [comment]
Restoring the Manager from backup can get complicated depending on how you are doing it. For example, if you are rebuilding a Manager node and using the tar file to restore it -AND- everything is the same: IP, hostname, interface name, etc., then the tar file should be fine, you would still want to regenerate certificates on the Manager and grid nodes. 

If it is just a restore from backup, say to data loss, then the tar file will be fine as long as the default path is still intact as well. This also makes the assumption you are restoring the same version of SO2.

If something is different, IP, hostname, interface name, etc., there would be more to edit to bring it back up. 

In what scenario are you looking to restore from backup?

**Reply by thegreekman** [reply]
Hello, I am trying to do a backup in the case of a catastrophic failure of the manager node. What would be the process to recover the data in such a case?(what needs to be backed up, how do you apply the backup)

**Reply by dougburks** [reply]
First, please see our backup documentation:
https://docs.securityonion.net/en/2.3/backups.html

Then review the above overview of the restore process:
https://github.com/Security-Onion-Solutions/securityonion/discussions/7859#discussioncomment-2680278

For more information, you can find the backup script on your system at `/usr/sbin/so-config-backup` and also at:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/so-config-backup

Finally, the tar man page may come in handy:
https://man7.org/linux/man-pages/man1/tar.1.html

**Reply by thegreekman** [reply]
So its just as simple as spinning up an identical manager node and untarring the backup/restoring the snapshot
